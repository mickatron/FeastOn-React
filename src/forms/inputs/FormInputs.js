import React from 'react';
import Switch from './Switch/Switch';
import TextArea from './TextArea/TextArea';

import HtmlInput from './HtmlInput/HtmlInput';

function renderHtmlInput({ ...props }) {
  return <HtmlInput {...props} />;
};

const TextInput = (props) => renderHtmlInput({ ...props, type: 'text' });
const NumberInput = (props) => renderHtmlInput({ ...props, type: 'number' });
const EmailInput = (props) => renderHtmlInput({ ...props, type: 'email' });
const PasswordInput = (props) => renderHtmlInput({ ...props, type: 'password' });
const CheckBox = (props) => renderHtmlInput({ ...props, type: 'checkbox' });
const Radio = (props) => renderHtmlInput({ ...props, type: 'radio' });

export {
  Switch,
  TextArea,
  TextInput,
  PasswordInput,
  NumberInput,
  EmailInput,
  CheckBox,
  Radio,  
};