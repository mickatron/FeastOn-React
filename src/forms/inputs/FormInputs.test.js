import { shallow } from 'enzyme';
import React from 'react';
import * as FormInputs from './FormInputs';

describe('FormInputs', () => {

  let component = null;
  describe('renders', () => {
    it('TextInput', () => {
      const ComponentToRender = FormInputs.TextInput;
      component = shallow(
        <ComponentToRender
          className="test"
          value="some test value"
        />
      );
      expect(component).toMatchSnapshot();
    });

    it('TextInput', () => {
      const ComponentToRender = FormInputs.NumberInput;
      component = shallow(
        <ComponentToRender
          className="test"
          value="some test value"
        />
      );
      expect(component).toMatchSnapshot();
    });

    it('TextInput', () => {
      const ComponentToRender = FormInputs.EmailInput;
      component = shallow(
        <ComponentToRender
          className="test"
          value="some test value"
        />
      );
      expect(component).toMatchSnapshot();
    });
  });

});