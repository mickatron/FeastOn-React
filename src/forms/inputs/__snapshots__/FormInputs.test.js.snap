// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormInputs renders TextInput 1`] = `
<HtmlInput
  className="test"
  type="text"
  value="some test value"
/>
`;

exports[`FormInputs renders TextInput 2`] = `
<HtmlInput
  className="test"
  type="number"
  value="some test value"
/>
`;

exports[`FormInputs renders TextInput 3`] = `
<HtmlInput
  className="test"
  type="email"
  value="some test value"
/>
`;
